### ISSUE TITLE
Mac - Issue in app quit

### ISSUE THREAD

**Issue by siddhartha-gupta** [nwjs/nw.js#3740]
Hi,

I'm facing an issue in quitting Mac application, with following setup and steps

- nw v0.12.3, with mac menu bar enabled with all options(edit, window etc)
- Using grunt node webkit builder to package the app
- Launch the app
- minimize the app to doc
- App menu is still visible and accessible
- Try to quit the app by pressing 'cmd+q' or bu clicking 'quit' from menu
- App is still running and in fact 'close' event is also not fired

**Comment by jgoalby** [comment]
Did you see this issue?
    https://github.com/nwjs/nw.js/issues/3162

If you did, please comment on how the resolution/workaround didn't work for you.  If you didn't see it, then I recommend searching the issues before creating a new issue.

If it is the same or very similar issue please consider closing this one.

**Comment by siddhartha-gupta** [comment]
After binding the 'shortcut' key as mentioned in #3162 I was able to receive the quit event on app minimize, but the event is propagated to application even when keys are pressed for some other application as well. Basically, I want to receive the quit event when 'cmd+q' is pressed for nw app only, and not for other apps. Is there a way to distinguish this?

**Comment by siddhartha-gupta** [comment]
Hi,

Can someone kindle provide an update/solution for the same :)

**Comment by jgoalby** [comment]
I also see the same issue.  It doesn't appear that #430 works any more in v12.  Not sure if something changed.  Doing CMD-TAB and CMD-Q on the icon works, but CMD-Q when in the dock doesn't work.  Also clicking quit in the menu when in the dock doesn't work.

I put a console log in the win.on("close" ...) and see it when not in the dock just fine.  Never see it when in dock except when CMD-TABing.

**Comment by siddhartha-gupta** [comment]
Yup, it seems close event is not propagated to app in case app is not in focus.

**Comment by siddhartha-gupta** [comment]
Any update?

**Comment by siddhartha-gupta** [comment]
Just a gentle bump!

**Comment by sporsh** [comment]
Just tested this in v0.15.2, and it's still an issue. Would love to see a fix for this!

**Comment by Christywl** [comment]
I can reproduce this issue on Mac with nwjs-sdk-v0.25.0.
